stdlib/atexit.c

```diff
@@ -34,7 +34,39 @@
 
 #include <stdlib.h>
 #include <dso_handle.h>
+#include <assert.h>
+#include <stdint.h>
+
+#include <libc-lock.h>
 #include "exit.h"
+#include <pointer_guard.h>
+
+int
+attribute_hidden
+__internal_atexit_2 (void (*func) (void),
+		   struct exit_function_list **listp)
+{
+  struct exit_function *new;
+
+  /* As a QoI issue we detect NULL early with an assertion instead
```

Comment:
What is QoI?

Reply:
I am also not sure. This function is copied from __internal_atexit, and I only changed its logic at the end: `new->func.at = func; new->flavor = ef_at;`. The comment was from original __internal_atexit